pkg/operator/ceph/client/controller_test.go

```diff
@@ -535,3 +552,257 @@ func TestReconcileCephClient_reconcileCephClientSecret(t *testing.T) {
+		assert.NoError(t, err)
+		assert.Contains(t, secret.StringData["userKey"], userKeyValue)
+	})
+
+	t.Run("brownfield reconcile - retain unknown cephx status", func(t *testing.T) {
+		cephClient := &cephv1.CephClient{}
+		err = cl.Get(ctx, req.NamespacedName, cephClient)
+		assert.NoError(t, err)
+		cephClient.Status.Cephx = cephv1.CephxStatus{}
+		err = cl.Update(ctx, cephClient)
+		assert.NoError(t, err)
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+
+		err = cl.Get(ctx, req.NamespacedName, cephClient)
+		assert.NoError(t, err)
+		assert.Equal(t, cephv1.CephxStatus{}, cephClient.Status.Cephx)
+
+		// get the secret and check the keyring
+		secret, err := clientset.CoreV1().Secrets(namespace).Get(ctx, "testSecret", metav1.GetOptions{})
+		assert.NoError(t, err)
+		assert.Contains(t, secret.StringData["userKey"], userKeyValue)
+	})
+
+	t.Run("rotate key - brownfield unknown status becomes known", func(t *testing.T) {
+		cephClient := &cephv1.CephClient{}
+		err = cl.Get(ctx, req.NamespacedName, cephClient)
+		assert.NoError(t, err)
+		cephClient.Spec.Security.CephX = cephv1.CephxConfig{
+			KeyRotationPolicy: "KeyGeneration",
+			KeyGeneration:     2,
+		}
+		err = cl.Update(ctx, cephClient)
+		assert.NoError(t, err)
+
+		rotatedKeyJson = `[{"key":"BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB=="}]`
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+
+		err = cl.Get(ctx, req.NamespacedName, cephClient)
+		assert.NoError(t, err)
+		assert.Equal(t, uint32(2), cephClient.Status.Cephx.KeyGeneration)
+		assert.Equal(t, "20.2.0-0", cephClient.Status.Cephx.KeyCephVersion)
+
+		// get the secret and check the keyring
+		secret, err := clientset.CoreV1().Secrets(namespace).Get(ctx, "testSecret", metav1.GetOptions{})
+		assert.NoError(t, err)
+		assert.Contains(t, secret.StringData["userKey"], "BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB==")
+	})
+
+	t.Run("brownfield reconcile - no further rotation happens", func(t *testing.T) {
+		// if rotation happens when it shouldn't, this will let us know by later comparison
+		rotatedKeyJson = `[{"key":"CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC=="}]`
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+
+		cephClient := &cephv1.CephClient{}
+		err = cl.Get(ctx, req.NamespacedName, cephClient)
+		assert.NoError(t, err)
+		assert.Equal(t, uint32(2), cephClient.Status.Cephx.KeyGeneration)
+		assert.Equal(t, "20.2.0-0", cephClient.Status.Cephx.KeyCephVersion)
+
+		// get the secret and check the keyring
+		secret, err := clientset.CoreV1().Secrets(namespace).Get(ctx, "testSecret", metav1.GetOptions{})
+		assert.NoError(t, err)
+		assert.NotContains(t, secret.StringData["userKey"], "CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC==")
+	})
+
+	t.Run("rotate key - cephx status updated", func(t *testing.T) {
```

Comment:
@BlaineEXE either this test work, or one above it work, 
Adding both together is keeping the reconcile not return.